docs-src/design-decisions.md

```diff
@@ -0,0 +1,44 @@
+# Design Decisions
+
+Throughout the design of these APIs some significant design decisions have been
+made. This provides some background on the factors considered when making these
+decisions.
+
+## GatewayClass Scope
+
+### Problem
+There were a number of compelling reasons to make GatewayClass a
+namespace-scoped resource. 
+
+### Advantages of a Namespace-Scoped GatewayClass
+* Less potential for naming conflicts among different implementations.
```

Comment:
I see this as a non-issue.  Classes should (ideally) be named qualitatively and abstractly, and are offered by a single "authority".

Reply:
That's a fair point, I've removed this.